model/src/main/java/de/fraunhofer/iosb/ilt/faaast/service/model/request/OutputModifierConstraints.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (c) 2021 Fraunhofer IOSB, eine rechtlich nicht selbstaendige
+ * Einrichtung der Fraunhofer-Gesellschaft zur Foerderung der angewandten
+ * Forschung e.V.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package de.fraunhofer.iosb.ilt.faaast.service.model.request;
+
+import de.fraunhofer.iosb.ilt.faaast.service.model.api.modifier.Content;
+import io.adminshell.aas.v3.model.builder.ExtendableBuilder;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+
+/**
+ * Allows defining constaints on which
+ * {@link de.fraunhofer.iosb.ilt.faaast.service.model.api.modifier.OutputModifier}
+ * are supported/accepted.
+ */
+public class OutputModifierConstraints {
+
+    private boolean supportsExtent;
+    private boolean supportsLevel;
+    private Set<Content> supportedContentModifiers;
+    public static final OutputModifierConstraints DEFAULT = new OutputModifierConstraints(true, false, Content.values());
+    public static final OutputModifierConstraints NONE = new OutputModifierConstraints(false, false);
+    public static final OutputModifierConstraints SUBMODEL = new OutputModifierConstraints(true, true, Content.NORMAL, Content.VALUE, Content.REFERENCE, Content.PATH);
+    public static final OutputModifierConstraints SUBMODEL_ELEMENT = new OutputModifierConstraints(true, true, Content.NORMAL, Content.VALUE, Content.REFERENCE, Content.PATH);
+    public static final OutputModifierConstraints ASSET_ADMINISTRATION_SHELL = new OutputModifierConstraints(false, false, Content.NORMAL, Content.REFERENCE);
+
+    public OutputModifierConstraints() {
```

Comment:
May add javadoc with hints for `this.supportsExtent = true` and `this.supportsLevel = false`